Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -66,13 +67,15 @@ final class TransactionPoster: TransactionPosterType {
     init(
         productsManager: ProductsManagerType,
         receiptFetcher: ReceiptFetcher,
+        transactionFetcher: StoreKit2TransactionFetcherType,
```

Comment:
This type keeps getting more dependencies 😅
I'm thinking we could wrap `ReceiptFetcher` and `StoreKit2ReceiptFetcher` into a common type, but I think that's probably more complex than it's worth it. But it is weird that `TransactionPoster` now needs a way to also fetch transactions :/